app/javascript/elm/src/Main.elm

```diff
@@ -67,6 +68,7 @@ type alias Model =
     , overlay : Overlay.Model
     , scrollPosition : Float
     , debouncingCounter : Int
+    , isNavExpanded : Bool
```

Comment:
I'd encourage as an experiment to create custom types also when a boolean would be enough. In this case
```
type Nav
  = Expanded
  | Collapsed
```

It's more code but it makes it more readable. Especially cause then stuff like
```
not model.isNavExpanded
```
becomes
```
model.nav == Collapsed
```
Also, this makes impossible to use `Nav` as a boolean for example when calling a function

Reply:
it doesn't seem to work like that :(

Reply:
I'm available next week if you want to take a look at this together ;)

Reply:
Can I join you? Cause I had a similar dilemma recently. I wanted to use the custom type but realized it will make an update branch longer. Instead of
```
ToggleSth ->
    (  { model | sth = not model.sth }, Cmd.none)
```
we would need:
```
ToggleSth ->
    let
        newSth = 
            case model.sth of
                 Value1 ->
                      Value2
                 Value2 ->
                      Value1

    (  { model | sth = newSth }, Cmd.none )
```
Or is there a shorter way? I thought about creating a custom equivalent of `not` function for my type?

Reply:
Yeah. You could use something like https://package.elm-lang.org/packages/joneshf/elm-tagged/latest/ and then just using `Tagged.map` would be enough

Reply:
is that what you prefer?

Reply:
It looks cool. Plus the implementation is prolly a few lines of code. Powerful pattern but really lightweight. If anyone wants to dig more, that library is using "phantom types" to "tag" different values of the same type

Reply:
I'd be +1 if anybody added that